dlt/common/utils.py

```diff
@@ -661,3 +665,168 @@ def is_typeerror_due_to_wrong_call(exc: Exception, func: AnyFun) -> bool:
 removeprefix = getattr(
     str, "removeprefix", lambda s_, p_: s_[len(p_) :] if s_.startswith(p_) else s_
 )
+
+
+def query_is_complex(
```

Comment:
I think this should go into `dlt/common/libs/sqlglot.py`, this is where @zilto has put the sqlglot stuff on the transformations branch.